image/types/src/lib.rs

```diff
@@ -392,4 +392,15 @@ impl ImageTocEntry {
     pub fn image_size(&self) -> u32 {
         self.size
     }
+
+    pub fn overlaps(&self, other: &ImageTocEntry) -> bool {
```

Comment:
Please add a unit test along the lines of https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d0a1229d9f8cf39e9d61aafcb43b0755

Reply:
Added.